lib/services/task_list_service.dart

```diff
@@ -26,6 +27,58 @@ class TaskListService with LogMixin, ChangeCallbackProvider {
     return (await _taskListCrdt).values;
   }
 
+  Future<List<Task>> getTasksForList(TaskList taskList) async {
+    var filtered = <Task>[];
+
+    List allTasks = await tasks;
+    for (var i = 0; i < allTasks.length; i++) {
```

Comment:
I haven't used sembast yet, but maybe with [Filter.equals](https://pub.dev/documentation/sembast/latest/sembast/Filter/Filter.equals.html) you can query the tasks for the ones where the key "taskListID" matches the value "taskList.id".

Reply:
i couldn't figure out the proper key yet.. but maybe @PitchBlackCoffee  @JacobRoe could help?
would work with

```
var finder = Finder(
      filter: Filter.equals(KEY, taskList.id),
 );

var records = await _keyValueRepository.find(finder);
```

and find in key_value_repository.dart as

```
Future<dynamic> find(Finder finder) async {
  return await _store.find(_db!, finder: finder);
} 
```